File: 프로그래머스/2/42586. 기능개발/기능개발.cpp
#include <string>
#include <vector>

using namespace std;

vector<int> solution(vector<int> progresses, vector<int> speeds) {
    vector<int> answer;
    
    for(int i=0; i<progresses.size(); i++){
        progresses[i] = 100-progresses[i];
        progresses[i] = progresses[i] % speeds[i] == 0 ? progresses[i] / speeds[i] : progresses[i] / speeds[i] + 1;
    }
    
    int tmp = progresses[0], cnt = 1;
    for(int i=1; i<progresses.size(); i++){
        if(progresses[i] <= tmp)
            cnt++;
        else{
            tmp = progresses[i];
            answer.push_back(cnt);
            cnt = 1;
        }
    }
    
    int sum=0;
    for(int i=0; i<answer.size(); i++)
        sum += answer[i];
    
    if(sum != progresses.size())
        answer.push_back(cnt);
    
    return answer;
}

File: 프로그래머스/2/42586. 기능개발/기능개발.js
function solution(progresses, speeds) {
    var answer = [];
    
    let remain = progresses.map((p, index) => {
        return Math.ceil((100-p) / speeds[index]);
    });
    
    let s=remain[0], cnt=1;
    for(let i=1; i<remain.length; i++){
        if(s >= remain[i])
            cnt++;
        else{
            s=remain[i];
            answer.push(cnt);
            cnt=1;
        }
    }
    
    let sum=0;
    for(d in remain)
        sum += d;
    
    if(sum !== progresses.length)
        answer.push(cnt);
    
    return answer;
}